### PR TITLE
[CET-2916] Historical sync job statuses

### PR DESCRIPTION
**Pull request by nikhilunni**
## Change description
image
image

New table in settings page to see status of historical Backstage <> Cortex sync jobs.

## Type of change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Improvement (improves codebase without changing functionality)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: src/components/SettingsPage/SyncJobsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+};
+
+const formatStatus = (status: JobStatus) => {
+  switch (status) {
+    case JobStatus.Done:
+      return 'Done';
+    case JobStatus.Cancelled:
+      return 'Cancelled';
+    case JobStatus.TimedOut:
+      return 'Timed Out';
```
Comment (cristinabuenahora) [inline src/components/SettingsPage/SyncJobsTable.tsx:58]: nit: 

```suggestion
      return 'Timed out';
```

File: src/components/SettingsPage/SyncJobsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    case JobStatus.Failure:
+      return 'Failure';
+    case JobStatus.InProgress:
+      return 'In Progress';
```
Comment (cristinabuenahora) [inline src/components/SettingsPage/SyncJobsTable.tsx:62]: nit: 

```suggestion
      return 'In progress';
```

File: src/components/SettingsPage/SyncJobsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  return (
+    <>
+      <Table
+        title="Sync Job Statuses"
+        columns={columns}
+        options={{ search: false }}
+        data={rows}
+      />
+    </>
```
Comment (cristinabuenahora) [inline src/components/SettingsPage/SyncJobsTable.tsx:133]: nit: 

```suggestion
      <Table
        title="Sync Job Statuses"
        columns={columns}
        options={{ search: false }}
        data={rows}
      />
```

File: src/components/SettingsPage/SyncJobsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  const rows: SyncJobRow[] | undefined = useMemo(() => {
+    return jobsResponse?.jobs?.map(job => {
+      return {
+        status: job.status,
+        timestamp: job.dateCreated,
+      };
+    });
+  }, [jobsResponse]);
+
+  if (loading || rows === undefined) {
```
Comment (cristinabuenahora) [inline src/components/SettingsPage/SyncJobsTable.tsx:113]:
```suggestion
  if (loading || isNil(rows)) {
```

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@cortexapps/backstage-plugin",
-  "version": "2.0.5",
+  "version": "2.0.6",
```
Comment (FoodProduct) [inline package.json:3]: Seems like we're adding new functionality, so this should be a minor version bump
```suggestion
  "version": "2.1.0",
```

File: src/api/types.ts [hunk 6]
```diff
@@ -308,3 +308,20 @@ export interface UserPermissionsResponse {
...
+
+export interface Job {
+  status: JobStatus;
+  dateCreated: string;
+}
+
+export enum JobStatus {
+  InProgress = 'IN_PROGRESS',
+  Done = 'DONE',
+  Cancelled = 'CANCELLED',
```
Comment (FoodProduct) [inline src/api/types.ts:324]: This is AMERICA 🇺🇸 . Keep your British spelling away from here 😤 
```suggestion
  Canceled = 'CANCELED',
```

Comment (nikhilunni) [reply src/api/types.ts:324]: the colonization has spread far and wide in our codebase 😓, returning from the backend like this right now -- but will update the display text here!

Comment (FoodProduct) [reply src/api/types.ts:324]: Sad 😿  but alright

**Review by FoodProduct — DISMISSED** [review]

File: src/components/SettingsPage/SyncJobsTable.tsx [hunk 7]
```diff
@@ -53,13 +54,13 @@ const formatStatus = (status: JobStatus) => {
     case JobStatus.Done:
       return 'Done';
     case JobStatus.Cancelled:
```
Comment (cristinabuenahora) [inline src/components/SettingsPage/SyncJobsTable.tsx:56]: 😭 lol

File: src/components/SettingsPage/SyncJobsTable.tsx [hunk 8]
```diff
@@ -123,13 +124,11 @@ export const SyncJobsTable = () => {
...
-    <>
-      <Table
-        title="Sync Job Statuses"
-        columns={columns}
-        options={{ search: false }}
-        data={rows}
-      />
-    </>
+    <Table
+      title="Sync Job Statuses"
```
Comment (cristinabuenahora) [inline src/components/SettingsPage/SyncJobsTable.tsx:128]:
```suggestion
      title="Sync job statuses"
```

**Review by cristinabuenahora — APPROVED** [review]

**Review by FoodProduct — DISMISSED** [review]

**Pull request merged**